Sources/NostrSDK/Filter.swift

```diff
@@ -54,19 +63,55 @@ public struct Filter: Codable, Hashable, Equatable {
     ///   - kinds: a list of a kind numbers
     ///   - events: a list of event ids that are referenced in an "e" tag
     ///   - pubkeys: a list of pubkeys that are referenced in a "p" tag
+    ///   - tags: a list of tag values that are referenced by single English-alphabet letter tag names
     ///   - since: an integer unix timestamp, events must be newer than this to pass
     ///   - until: an integer unix timestamp, events must be older than this to pass
     ///   - limit: maximum number of events to be returned in the initial query
     ///
+    /// If `tags` contains an `e` tag and `events` is also provided, `events` takes precedence.
```

Comment:
I kept in `events` and `pubkeys` for backwards compatibility, but also developer ergonomics. Let me know if you think this is a bad idea.